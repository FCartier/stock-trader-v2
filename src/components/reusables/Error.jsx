import React from 'react'

const Error = props => {
    const { error } = props
    return (
        <div>
            {error && 
             <h2>Error, try another search</h2>
            }
        </div>
    )
}

export default Error;