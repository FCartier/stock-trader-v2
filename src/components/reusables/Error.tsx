import React from 'react';

interface ErrorProps {
    error : {
        message: string;
        stack: string;
    }
}

const Error: React.SFC<ErrorProps> = ({error}) => {
    return !(typeof error === "undefined" || error.message === "") ? (
        <div><h2>Error, try another search</h2></div>
    ) : null;
}

export default Error;