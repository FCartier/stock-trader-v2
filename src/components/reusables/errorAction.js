export const ERROR_ACTION = "ERROR ACTION";

export function errorAction(payload) {
  return {
    type: ERROR_ACTION,
    payload
  };
}