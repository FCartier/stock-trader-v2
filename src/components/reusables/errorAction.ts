import { Action, ActionCreator } from 'redux';

export const ERROR_ACTION = "ERROR_ACTION";
export type ERROR_ACTION = typeof ERROR_ACTION;

export interface ErrorState {
    message: string;
    stack: string;
}

export interface ErrorAction extends Action {
  type: typeof ERROR_ACTION
  error: ErrorState
}

export const errorAction: ActionCreator<ErrorAction> = (error: ErrorState) => ({
  type: ERROR_ACTION,
  error
})
