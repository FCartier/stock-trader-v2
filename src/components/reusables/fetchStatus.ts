export enum fetchStatus {
  pending,
  success,
  failed
}