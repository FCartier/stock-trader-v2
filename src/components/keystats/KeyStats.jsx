import React from "react";
import { SectionHeader } from "../reusables/SectionHeader";

const KeyStats = props => {
  const { keystats } = props;

  return Object.keys(keystats).length ? (
    <div>
        <div className="keystats">
          <SectionHeader title="KEY STATS" />
          <div className="keystats-row">
            <div className="keystats-column">
              <ul className="keystats-list">
                <li>
                  <p>Previous Close</p> <h5>{keystats.close}</h5>
                </li>
                <li>
                  <p>Day Range</p>{" "}
                  <h5>
                    {keystats.low}-{keystats.high}
                  </h5>
                </li>
                <li>
                  <p>Volume</p> <h5>{keystats.latestVolume}</h5>
                </li>
              </ul>
            </div>
            <div className="keystats-column">
              <ul className="keystats-list">
                <li>
                  <p>Open </p> <h5>{keystats.open}</h5>
                </li>
                <li>
                  <p>52 Week Range</p>{" "}
                  <h5>
                    {keystats.week52Low}-{keystats.week52High}
                  </h5>
                </li>
                <li>
                  <p> Total Avg. Volume</p> <h5>{keystats.avgTotalVolume}</h5>
                </li>
              </ul>
            </div>
          </div>
        </div>
    </div>
  ) : null
};

export default KeyStats;
