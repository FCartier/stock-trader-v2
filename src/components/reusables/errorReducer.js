import { ERROR_ACTION } from "./errorAction"

export const error = (state = {}, action) => {
    switch (action.type) {
      case ERROR_ACTION:
        return action.payload;
      default:
        return state;
    }
  };