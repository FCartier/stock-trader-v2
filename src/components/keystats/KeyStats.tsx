import * as React from "react";
import { SectionHeader } from "../reusables/SectionHeader";
import { Row, Column, List } from "./styles";
import { fetchStatus } from "../reusables/fetchStatus";
import NoData from "../reusables/NoData";

interface IKeyStatsProps {
  keystats?: {
    close: number;
    low: number;
    high: number;
    open: number;
    latestVolume: number;
    week52High: number;
    week52Low: number;
    avgTotalVolume: number;
  };
  status?: fetchStatus;
}

const KeyStats: React.SFC<IKeyStatsProps> = ({ keystats, status }) => {
  return status !== fetchStatus.pending ? (
    <div>
      <SectionHeader title="KEY STATS" />
      {status === fetchStatus.success ? (
        <Row>
          <Column>
            <List>
              <li>
                <p>Previous Close</p> <h5>{keystats.close}</h5>
              </li>
              <li>
                <p>Day Range</p>{" "}
                <h5>
                  {keystats.low}-{keystats.high}
                </h5>
              </li>
              <li>
                <p>Volume</p> <h5>{keystats.latestVolume}</h5>
              </li>
            </List>
          </Column>
          <Column>
            <List>
              <li>
                <p>Open </p> <h5>{keystats.open}</h5>
              </li>
              <li>
                <p>52 Week Range</p>{" "}
                <h5>
                  {keystats.week52Low}-{keystats.week52High}
                </h5>
              </li>
              <li>
                <p> Total Avg. Volume</p> <h5>{keystats.avgTotalVolume}</h5>
              </li>
            </List>
          </Column>
        </Row>
      ) : (
        <NoData componentName="key stats" />
      )}
    </div>
  ) : null;
};

export default KeyStats;
