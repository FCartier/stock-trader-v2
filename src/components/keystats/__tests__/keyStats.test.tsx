import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import KeyStats from "../KeyStats";
import { testData } from "../__data__/keyStats.data";
import { fetchStatus } from "../../reusables/fetchStatus";

configure({ adapter: new Adapter() });

describe("keystats component", () => {
  test("keystats component should render properly before fetch", () => {
    const component = shallow(
      <KeyStats
        keystats={{
          close: 0,
          low: 0,
          high: 0,
          open: 0,
          latestVolume: 0,
          week52High: 0,
          week52Low: 0,
          avgTotalVolume: 0
        }}
        status={fetchStatus.pending}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test("keystats component should render properly with fetch success", () => {
    const component = shallow(
      KeyStats({ keystats: testData, status: fetchStatus.success })
    );
    expect(component).toMatchSnapshot();
  });

  test("keystats component should render properly with fetch failed", () => {
    const component = shallow(
      KeyStats({ keystats: testData, status: fetchStatus.failed })
    );
    expect(component).toMatchSnapshot();
  });
});
