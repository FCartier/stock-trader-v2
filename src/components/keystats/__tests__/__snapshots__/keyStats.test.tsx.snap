// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keystats component keystats component should render properly before fetch 1`] = `""`;

exports[`keystats component keystats component should render properly with fetch failed 1`] = `
<div>
  <Styled(div)>
    <Component
      title="KEY STATS"
    />
    <NoData
      componentName="key stats"
    />
  </Styled(div)>
</div>
`;

exports[`keystats component keystats component should render properly with fetch success 1`] = `
<div>
  <Styled(div)>
    <Component
      title="KEY STATS"
    />
    <Styled(div)>
      <Styled(div)>
        <Styled(ul)>
          <li>
            <p>
              Previous Close
            </p>
             
            <h5>
              221.19
            </h5>
          </li>
          <li>
            <p>
              Day Range
            </p>
             
            <h5>
              219.34
              -
              222.64
            </h5>
          </li>
          <li>
            <p>
              Volume
            </p>
             
            <h5>
              22688651
            </h5>
          </li>
        </Styled(ul)>
      </Styled(div)>
      <Styled(div)>
        <Styled(ul)>
          <li>
            <p>
              Open 
            </p>
             
            <h5>
              222.15
            </h5>
          </li>
          <li>
            <p>
              52 Week Range
            </p>
             
            <h5>
              150.24
              -
              233.47
            </h5>
          </li>
          <li>
            <p>
               Total Avg. Volume
            </p>
             
            <h5>
              34502868
            </h5>
          </li>
        </Styled(ul)>
      </Styled(div)>
    </Styled(div)>
  </Styled(div)>
</div>
`;
